operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/DependentResource.java

```diff
@@ -68,4 +68,8 @@ static String defaultNameFor(Class<? extends DependentResource> dependentResourc
   default boolean isDeletable() {
     return this instanceof Deleter;
   }
+
+  String getName();
```

Comment:
Ad other approach would be just have 
`String name()`

And the setter in a separate Inteface, but we want to set it from spec.
What do you think @metacosm

Reply:
Changed it to a setter interface. Both are a bit ugly. But this highest level interface (`DependentResource`) is more consistent and readable.